### DISCUSSION TITLE
How can I set TextDecorations to a ContentPresenter?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10681] [Q&A] [answered]
I am creating a TemplatedControl base on Button, here is my code:
` <ControlTemplate>
			  <Border Name="B" Background="{TemplateBinding Background}" BorderBrush="{TemplateBinding BorderBrush}" BorderThickness="{TemplateBinding BorderThickness}">
				  <ContentPresenter Margin="{TemplateBinding Padding}" HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}" VerticalAlignment="{TemplateBinding VerticalContentAlignment}" Content="{TemplateBinding Content}"></ContentPresenter>
			  </Border>
		  </ControlTemplate>`

In the button there is a TextDecorations property, I want to TemplateBind it to the ContentPresenter. When changing the TextDecorations property, the TextDecorations of Text in the ContentPresent will change also.

How can I achieve this?

**Comment by timunie** [comment, marked as answer]
@kyuranger which Avalonia version do you use?

**Reply by kyuranger** [reply]
> @kyuranger which Avalonia version do you use?

V11.0.0 Preview5, thank you.

**Reply by timunie** [reply]
In that case it is `TextElement.FontSize` and simiar attached properties you have to set. They can also be template bound.

**Reply by kyuranger** [reply]
I set my code like this:
`<ContentPresenter Name="CP" TextElement.TextDecorations="{TemplateBinding TextDecorations}"  Margin="{TemplateBinding Padding}" HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}" VerticalAlignment="{TemplateBinding VerticalContentAlignment}" Content="{TemplateBinding Content}"></ContentPresenter>`
And VS reports that TextElement does not contain a TextDecorations property:
Unable to resolve suitable regular or attached property TextDecorations on type Avalonia.Controls:Avalonia.Controls.Documents.TextElement Line 20, position 8.

In https://reference.avaloniaui.net/api/ I can't find any about the TextElement that I can not find any ways to solve this.

**Reply by Gillibald** [reply]
TextDecorations is not an attached property and only available on TextBlock itself. Just define a content template with a TextBlock that sets TextDecorations via binding etc.

**Reply by kyuranger** [reply]
> TextDecorations is not an attached property and only available on TextBlock itself. Just define a content template with a TextBlock that sets TextDecorations via binding etc.

All right, it seems I have to do it like this.

**Reply by timunie** [reply]
Probably a Style for TextBlock can also work, if you want to be applied on all child TextBlocks

**Reply by kyuranger** [reply]
> Probably a Style for TextBlock can also work, if you want to be applied on all child TextBlocks

I only found how to achieve this in WPF:https://stackoverflow.com/questions/13628101/how-do-i-correctly-apply-a-style-to-content-presenter

But I don't know how to code it in Avalonia.

**Reply by timunie** [reply]
```xml
<MyControl.Styles>
   <Style Selector="TextBlock">
      <Setter Property="..." Value="..." />
    </Style>
</MyControl.Styled>
```

Or 

```xml
<App.Styles>
   <Style Selector="MyControl TextBlock"> 
    [ ... See above ... ]
```

https://docs.avaloniaui.net/docs/styling/selectors

**Reply by kyuranger** [reply]
> ```
> <MyControl.Styles>
>    <Style Selector="TextBlock">
>       <Setter Property="..." Value="..." />
>     </Style>
> </MyControl.Styled>
> ```
> 
> Or
> 
> ```
> <App.Styles>
>    <Style Selector="MyControl TextBlock"> 
>     [ ... See above ... ]
> ```
> 
> https://docs.avaloniaui.net/docs/styling/selectors

Let me try this. Thank you so much.
